script/script_test.go

```diff
@@ -42,6 +44,38 @@ func (l *luaContext) ParentSpan() opentracing.Span          { return nil }
 func (l *luaContext) Split() (filters.FilterContext, error) { return nil, nil }
 func (l *luaContext) Loopback()                             {}
 
+func (l *luaContext) Logger() *log.Entry {
+	traceId := l.getTraceID()
+	return log.WithFields(log.Fields{"trace_id": traceId})
+}
+
+func (l *luaContext) getTraceID() string {
+	if l == nil {
+		return "(nil context)"
+	}
+	if l.request == nil {
+		return "(nil request)"
+	}
+
+	span := opentracing.SpanFromContext(l.request.Context())
+	if span == nil {
+		return "(nil span)"
+	}
+
+	spanContext := span.Context()
+	if spanContext == nil {
+		return "(nil span context)"
+	}
+
+	switch spanContextType := spanContext.(type) {
+	case lightstep.SpanContext:
+		_ = spanContextType
+		return fmt.Sprintf("0x%x", spanContext.(lightstep.SpanContext).TraceID)
+	}
+
+	return "(unknown span context type)"
+}
```

Comment:
It looks like we can create a helper (maybe in tracing package) to extract traceId from http.Request and reuse it in all filterContext implementations.